### PR TITLE
Vet center - "Nearby locations" should be retrieved from Lighthouse if not published

### LINKED ISSUE #24312: Nonpublished "Nearby locations" should be retrieved from Lighthouse (Vet Centers)
## Issue Description
Although we have used the process for a few situations, unpublished content generally should not be rendered on the front end.  Because Vet Center pages are early in the rollout, it is likely that the facilities included in the "nearby locations" section of an MVP page are still **unpublished**. 

We need to change the logic and retrieve data from Lighthouse as the source of truth for the facility until a site is published. 

---
## Tasks
- [ ] Add check to see if the node referenced in `field_nearby_vet_centers` is published. 
      - If **no**, get `field_facility_id` to retrieve the data from Lighthouse (similar to VAMC pages)
      - if **yes**, use current process

## Acceptance Criteria
- [ ] Lighthouse is the source of truth for unpublished facilities referenced in Vet Center "Nearby locations" lists 
- [ ] List displayed does not contain duplicates
- [ ] Location is included whether or not it is a published location

### PR DESCRIPTION
**Pull request by brangi**
resolves https://github.com/department-of-veterans-affairs/va.gov-team/issues/24312

### CODE REVIEW

**Comment by mpelzsherman** [comment]
Much of this code duplicates the display logic in the liquid templates.

Do we have a plan to clean up that duplication so we don't have to maintain the same code in two places?

**Comment by brangi** [comment]
> Much of this code duplicates the display logic in the liquid templates.
> 
> Do we have a plan to clean up that duplication so we don't have to maintain the same code in two places?

I would love that, maybe make everything dynamic ? however from conversations I have had with @mmiddaugh, we still have to look at the two sources cms and lighthouse, specially for unpublished content. I hate this implementation tbh but the requirement is still there....

File: src/applications/static-pages/facilities/vet-center/components/VetCenterInfoSection.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        </div>
+      </div>
+      <h4 className="force-small-header vads-u-margin-top--0 vads-u-line-height--1 vads-u-margin-bottom--1">
+        Phone
+      </h4>
+      <div>
+        {props.vetCenter.entityBundle === 'vet_center_cap'
+          ? renderPhone(props.mainVetCenterPhone)
+          : renderPhone(props.vetCenter.fieldPhoneNumber)}
+      </div>
```
Comment (mpelzsherman) [inline src/applications/static-pages/facilities/vet-center/components/VetCenterInfoSection.jsx:67]: Should we check for phone data before rendering the `div` tags?

**Review by mpelzsherman — APPROVED** [review]
I'd like to see some unit tests added for these new components at some point.

**Review by jhonny218 — APPROVED** [review]

**Pull request merged**